### PR TITLE
Add support for PipelineRun minimal embedded status

### LINKED ISSUE #1618: Support PipelineRun minimal embedded status
Starting in Pipeline v0.35.0, we support a new approach to embedding `TaskRun` and `Run` statuses in `PipelineRun.Status`, as described in [TEP-0100](https://github.com/tektoncd/community/blob/main/teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md). When the `embedded-status` feature flag is set to `minimal`, the `PipelineRun.Status.TaskRuns` and `PipelineRun.Status.Runs` fields aren't populated. Instead, a new field, `PipelineRun.Status.ChildReferences` is populated with a list of the API version, kind, name, and `PipelineTask` name for each `TaskRun` or `Run` in the `PipelineRun`, requiring anything trying to see the status of the children of a `PipelineRun` to fetch those statuses directly.

The CLI needs to be updated to support minimal embedded status in the next few months, since the default will be switching to `minimal` in Pipeline v0.44.0 and the feature flag will be going away in Pipeline v0.45.0.

This is a requirement for https://github.com/tektoncd/pipeline/issues/4954

/kind feature
/kind tep

### PR DESCRIPTION
**Pull request by abayer**
# Changes

fixes #1618 

This is a dependency for https://github.com/tektoncd/pipeline/issues/4954

Starting in Pipeline v0.35.0, we support a new approach to embedding `TaskRun` and `Run` statuses in `PipelineRun.Status`, as described in [TEP-0100](https://github.com/tektoncd/community/blob/main/teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md). When the `embedded-status` feature flag is set to `minimal`, the `PipelineRun.Status.TaskRuns` and `PipelineRun.Status.Runs` fields aren't populated. Instead, a new field, `PipelineRun.Status.ChildReferences` is populated with a list of the API version, kind, name, and `PipelineTask` name for each `TaskRun` or `Run` in the `PipelineRun`, requiring anything trying to see the status of the children of a `PipelineRun` to fetch those statuses directly.

Until Pipeline v0.44.0, the default value for the `embedded-status` feature flag is `full`, continuing the previous full embedded status behavior, but in v0.44.0, the default will be `minimal`, and in v0.45.0, the feature flag will go away, and the minimal embedded status will always be used. Therefore, the CLI needs to add support for minimal embedded status ASAP.

This change uses a helper function added in Pipeline v0.36.0 (which is just copied into the CLI code for now due to Chains not being compatible with Pipeline v0.36.0) to take a `PipelineRun`, check if it already has the deprecated `TaskRuns` and `Runs` maps populated (which will be the case for Pipeline pre-v0.35.0, and later Pipeline versions if the `embedded-status` feature flag is either `full` or `both`), returning those maps if they are populated, and otherwise generating those maps from the `PipelineRun`'s `ChildReferences` and returning the resulting maps. The functions used in the CLI to fetch `PipelineRun`s will now call that helper function and set `PipelineRun.Status.TaskRuns` and `PipelineRun.Status.Runs` to the output of that function, ensuring that no other code changes should be needed in the CLI until those fields are removed.

/kind feature
/kind tep

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
Add support for Pipeline's minimal embedded status feature flag.
```

### CODE REVIEW

**Comment by abayer** [comment]
cc @vdemeester @afrittoli @jerop

**Comment by jerop** [comment]
@vinamra28 please take a look 🙏🏾

**Review by vdemeester — APPROVED** [review]

File: pkg/cmd/pipelinerun/logs_test.go [hunk 1]
```diff
@@ -347,24 +348,43 @@ func TestPipelinerunLogs(t *testing.T) {
...
+						}, {
+							Name:             tr2Name,
+							PipelineTaskName: task2Name,
+							TypeMeta: runtime.TypeMeta{
+								APIVersion: "tekton.dev/v1beta1",
+								Kind:       "TaskRun",
+							},
+						},
 					},
+					/*
```
Comment (pradeepitm12) [inline pkg/cmd/pipelinerun/logs_test.go:382]: may be we can remove this.

File: pkg/pipelinerun/pipelinerun.go [hunk 2]
```diff
@@ -209,3 +229,71 @@ func createUnstructured(obj runtime.Object, c *cli.Clients, opts metav1.CreateOp
...
+		fmt.Fprintf(os.Stderr, "Failed to get TaskRun and Run statuses for PipelineRun %s from namespace %s\n", pr.Name, ns)
+		return nil, err
+	}
+	pr.Status.TaskRuns = taskRunMap
+	pr.Status.Runs = runMap
+
+	return &pr, nil
+}
+
+// getFullPipelineTaskStatuses returns populated TaskRun and Run status maps for a PipelineRun from its ChildReferences.
```
Comment (pradeepitm12) [inline pkg/pipelinerun/pipelinerun.go:245]: :+1:

File: test/e2e-tests.sh [hunk 3]
```diff
@@ -172,4 +178,13 @@ fi
 go_test_e2e ./test/e2e/... || failed=1
 (( failed )) && fail_test
 
+# Re-run the PipelineRun tests with "embedded-status" set to "minimal"
+header "Running Go e2e tests with Pipeline embedded-status set to minimal"
+
+set_minimal_embedded_status
```
Comment (afrittoli) [inline test/e2e-tests.sh:184]: Should we keep testing the old behaviour too? Not sure how though, I guess that would mean an extra e2e job?

Comment (abayer) [reply test/e2e-tests.sh:184]: We are still testing that - we're running the full set of tests with the default (`full`), and then updating the flag to `minimal` and rerunning the relevant tests.

Comment (abayer) [reply test/e2e-tests.sh:184]: I decided it wasn't worth adding an additional job given how few tests actually change behavior depending on the `embedded-status` value - it was markedly simpler and faster to get what we needed by just doing it this way. =)

**Pull request merged**